game_input/src/lib.rs

```diff
@@ -1,22 +1,21 @@
 extern crate game_window;
 extern crate game_logic;
 
-use std::iter::IteratorExt;
 use game_logic::LogicState;
 use game_window::{Action, Key, WindowEvent};
 
 pub type InputHandler = fn(LogicState, WindowEvent) -> LogicState;
 
-pub fn handle_events(logic_state: LogicState, events: Vec<WindowEvent>) -> LogicState {
-  events.iter().fold(logic_state, check_keys)
-}
-
-fn check_keys(logic_state: LogicState, event: &WindowEvent) -> LogicState {
-  match *event {
-    WindowEvent::Key(Key::Escape, _, Action::Press, _) => logic_state.as_closed(),
-    WindowEvent::Key(Key::A, _, Action::Press, _) => logic_state.inc_edges(),
-    WindowEvent::Key(Key::D, _, Action::Press, _) => logic_state.dec_edges(),
-    _ => logic_state
+pub fn handle_events(logic_state: &mut LogicState, events: Vec<WindowEvent>) {
+  for event in events.iter() {
+    match *event {
+      WindowEvent::Key(Key::Escape, _, Action::Press, _) => logic_state.closed(),
+      WindowEvent::Key(Key::A, _, Action::Press, _) => logic_state.inc_edges(),
+      WindowEvent::Key(Key::D, _, Action::Press, _) => logic_state.dec_edges(),
+      WindowEvent::Key(Key::A, _, Action::Repeat, _) => logic_state.inc_edges(),
+      WindowEvent::Key(Key::D, _, Action::Repeat, _) => logic_state.dec_edges(),
+      _ => ()
+    }
```

Comment:
I think the role of LogicState really needs to be considered more thoroughly. As it picks up more responsibility, I'm not sure I feel great about all the `clone()`s. 

Will think on it more.